src/util/TypeTraits.h

```diff
@@ -215,4 +215,47 @@ template <typename... Ts>
 requires(sizeof...(Ts) > 0)
 using First = typename detail::FirstWrapper<Ts...>::type;
 
+/*
+@brief Require `Fn` to be invocable with `Args...` and the return type to be
+`isSimilar` with `Ret`.
+*/
+template <typename Fn, typename Ret, typename... Args>
+concept InvocableWithSimilarReturnType =
+    std::invocable<Fn, Args...> &&
+    isSimilar<std::invoke_result_t<Fn, Args...>, Ret>;
+
+/*
+@brief Require `Fn` to be invocable with `Args...` and the return type to be
+ `Ret`.
+*/
+template <typename Fn, typename Ret, typename... Args>
+concept InvocableWithExactReturnType =
+    std::invocable<Fn, Args...> &&
+    std::same_as<std::invoke_result_t<Fn, Args...>, Ret>;
+
+/*
+@brief Require `Fn` to be regular invocable with `Args...` and the return type
+to be `isSimilar` with `Ret`.
+
+Note: Currently, the difference between invocable and regular invocable is
+purely semantic. In other words, we can not, currently, actually check, if an
+invocable type is regular invocable, or not.
```

Comment:
Maybe add a link to the `cppreference` site for `std::invocable` for details.

Reply:
Done, see [80acf94](https://github.com/ad-freiburg/qlever/pull/1118/commits/80acf941d943fabb1d27aa684d4606eb2ac514b9).